src/ansiblelint/rules/args.py

```diff
@@ -73,8 +73,9 @@ class ValidationPassedError(Exception):
 class CustomAnsibleModule(basic.AnsibleModule):  # type: ignore[misc]
     """Mock AnsibleModule class."""
 
-    def __init__(self, *args: str, **kwargs: str) -> None:
+    def __init__(self, *args: str, **kwargs: Any) -> None:
         """Initialize AnsibleModule mock."""
+        kwargs["no_log"] = True
         super().__init__(*args, **kwargs)
```

Comment:
I don't really like the idea of _introducing_ `Any` anywhere if we can help it. We could make the type `str | bool` instead, assuming that previously we are only ever calling it with strings.

```suggestion
    def __init__(self, *args: str, **kwargs: str | bool) -> None:
        """Initialize AnsibleModule mock."""
        kwargs["no_log"] = True
        super().__init__(*args, **kwargs)
```

We could instead keep the hint but instead `kwargs.pop("no_log")` and then pass `no_log=True` directly to `super().__init__()`, but I'm not convinced `str` actually _is_ the correct type here

Reply:
kwargs and type is complex and bit unexpected, for the time being "Any" is the correct solution. As you can see mypy does not complain about it.